hazelcast/src/main/java/com/hazelcast/map/impl/event/MapEventPublisherImpl.java

```diff
@@ -119,7 +119,11 @@ public void publishWanRemove(String mapName, Data key) {
     protected void publishWanEvent(String mapName, ReplicationEventObject event) {
         MapContainer mapContainer = mapServiceContext.getMapContainer(mapName);
         WanReplicationPublisher wanReplicationPublisher = mapContainer.getWanReplicationPublisher();
-        wanReplicationPublisher.publishReplicationEvent(SERVICE_NAME, event);
+        if (isOwnedPartition(event.getKey())) {
+            wanReplicationPublisher.publishReplicationEvent(SERVICE_NAME, event);
+        } else {
+            wanReplicationPublisher.publishReplicationEventBackup(SERVICE_NAME, event);
```

Comment:
A thing that I've recently noticed is that we didn't (and still don't I guess) support backup events in OS code. The methods calling `publishWanEvent` are actually returning if the event key is not local. So this part is unnecessary.

Reply:
Ah, ok, this method is used by EE as well. Ignore my comment.